awscli/customizations/history/db.py

```diff
@@ -83,11 +79,53 @@ def _encode_mutable_mapping(self, obj):
     def _encode_datetime(self, obj):
         return obj.isoformat()
 
+    def _try_encode(self, obj):
+        try:
+            obj = obj.decode('utf-8')
+        except UnicodeDecodeError:
+            obj = '<Byte sequence>'
+        return obj
+
+    def _remove_non_unicode_stings(self, obj):
+        if isinstance(obj, str):
+            obj = self._try_encode(obj)
+        elif isinstance(obj, dict):
```

Comment:
Hmm. Would it be possible to do this in place or would does that risk mutating the payload in the upstream code logic?

Reply:
That would probably mutate it. I added a few tests to check for mutation.